app/Http/Controllers/api/StudentDetailController.php

```diff
@@ -1,15 +1,31 @@
 <?php
-
+declare(strict_types=1); 
 namespace App\Http\Controllers\api;
 
 use App\Http\Controllers\Controller;
+use App\Models\Resume;
+use App\Service\StudentDetailsService;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use App\Exceptions\StudentNotFoundException;
 
 class StudentDetailController extends Controller
 {
-    public function __invoke()
-    {
-        $data = file_get_contents(base_path('database/data/studentDetail.json'));
+    private StudentDetailsService $studentDetailsService;
 
-        return response()->json(json_decode($data, true));
+    public function __construct(StudentDetailsService $studentDetailsService)
+    {
+        $this->studentDetailsService =$studentDetailsService;
     }
+    function __invoke(Request $request,$student):JsonResponse
+    {
+        try {
+            $service = $this->studentDetailsService->execute($student);
+            return response()->json(['data'=> [$service]], 200);
```

Comment:
If you want, you can omit the 200 code here.

Reply:
I think I'll leave it, it fits better in harmony with the rest of the code.